src/compiler/compile/render_dom/wrappers/InlineComponent/index.ts

```diff
@@ -393,7 +393,7 @@ export default class InlineComponentWrapper extends Wrapper {
 
 			component.partly_hoisted.push(body);
 
-			return b`@binding_callbacks.push(() => @bind(${this.var}, '${binding.name}', ${id}));`;
+			return b`@binding_callbacks.push(() => @bind(${this.var}, '${binding.name}', ${id}, ${snippet}));`;
```

Comment:
In what sorts of situations will `${snippet}` be an expression that is `undefined` vs non-`undefined`? Is this just an approximation of whether it's the first call or not, and we're accepting that `${snippet|` might be `undefined` some other times, at which point we'll call `callback` when ideally we wouldn't have to? Or is something else going on?

Reply:
In 
```html
<script>
  let foo = "bar";
</script>

<SomeConponent bind:foo />
```

snippet is the value of foo, so it's defined in this case as bar, and would be undefined if foo wasn't initialized, at which point we want to assign foo to the value that foo has inside SomeComponent.

I don't know why it worked before having the correct value for foo (being bar, not the value of foo inside SomeComponent), but having this check breaks the infinite loop in the related issues.